### ISSUE TITLE
TFC.tv login changed => Plugin doesn't work!

### ISSUE THREAD

**Issue by magnuslsjoberg** [magnuslsjoberg/The-Filipino-Channel.bundle#14]
TFC has changed their login again. This time it takes you to another web page and you need to login, choose a "Kapamiliya User Name" and verify your email address.  Unfortunately, I haven't received any confirmation email despite trying several times with different browsers and both my TFC Premium and a newly created TFC Free email address. 

I cant't debug anything until I manage to login at their web site.

Currently waiting for an agent at TFC Live Chat Support...

**Comment by Sapphica** [comment]
Did you find a fix for it?  Really miss it working...

**Comment by magnuslsjoberg** [comment]
Unfortunately, not!  I can’t even login at the TFC website with a browser or the iPhone app.  I spent 45 minutes chatting with TFC helpdesk yesterday and today I finally managed to get a verification email from them. So theoretically, I can login to TFC but in reality it hangs with the 5 colorful circles “Loading...” forever.

Until I can login the normal way I can’t do anything more... really annoyed right now since we normally watch TFC every night.
...and Plex is so far the best way to watch it!

A shame that Plex have decided to remove plugins in a near future so we may have to swap to Kodi.

The Kodi TFC plugin is of course experiencing the same problems but first of all they have had been able to login and secondly the developers for that plugin is way more skilled than me. I hope that I can take their solution and integrate in the Plex TFC plugin as soon as I am able to login myself!

**Comment by Drucio** [comment]
Yeah... Annoying but if ur a roku user TFC finally has an app

**Comment by Sapphica** [comment]
want to use my user name a password for TFC?  I had to update my account as well but didn't have any issues on win10 and edge. 

I hope you keep this plugin going, I know a few people that use it.
 =D

**Comment by magnuslsjoberg** [comment]
@Sapphica:

I registered two new accounts to try to login but they didn’t work either.  I also already got someone else’s login to test with but again it failed. 

I have cleared caches and cookies and tried on computer and iPhone both through my ISP and my 3G on the phone.

The intention is to keep the plugin alive for as long as possible (the family peace depends on it =] )

**Comment by stakker44** [comment]
magnuslsjoberg,

I am able to finally login as well after turning off privacy badger on my browser for a couple of redirects that they are doing with the site. Is there any specific info that you need? I can assist if you can walk me through it.

**Comment by ptfir** [comment]
Any fix yet? Kodi tfc.tv has also same problem. Currently they are working with user agent headers fix.

**Comment by magnuslsjoberg** [comment]
@ptfir:

I am following https://github.com/cmik/cmik.xbmc closely since they have been my rescue earlier on when I have had problems with the TFC website.

This time I am still struggling to just login!  I can login with the TFC iPhone app and watch but haven't been able to get TFC to work in a computer browser.

The Kodi plugin login sequence doesn't work for me so I am trying to find out one that works. Not easy with 2 or 3 websites involved in the session with
cookie settings, javascripts and all https which make it quite difficult to follow.

I still haven't given up though! =]

**Comment by magnuslsjoberg** [comment]
@ptfir:

No, the TFC Plex plugin can not login to TFC.tv!

You can browse all pages fine, but when you click on a video to start watching, the plugin must login to be able to retrieve the correct information
about the stream so that it can play it. Now you will just end up with some error message that the video can't be played or the server is not powerful enough
or something similar. (I agree that I could have put more effort into trying to display an error message rather than crash and let Plex handle.)

So, first I need to be able to login to extract the information, once that it is place I will see what other problems we may have!

**Comment by ptfir** [comment]
You're right it can browse but doesn't show any message if login is a success or failed. 👍

**Comment by nleoncio** [comment]
looks like the cmik plugin for kodi is still stuck as well. No working solution still. 

I tried sideloading the tfc app on my firestick but it doesn't play video contiuously. It will stop playing after a few minutes.

**Comment by magnuslsjoberg** [comment]
The good news is that I have been able to do a scripted login with ”normal” python code. 

The bad news is that Plex has some limitations and peculiarities so when I run the plugin on the Plex server it doesn’t work at all.

This shouldn’t be too difficult to fix but it is hard to find the time. I spent most of the weekend and two late nights to be able to come where I am and more work is needed.

Once I manage to get the login work in Plex, I still have to tackle the problems with the actual streams that Kodi plugin has...

I haven’t given up yet, but don’t expect a working plugin in the nearest future.

**Comment by magnuslsjoberg** [comment]
Ok, almost given up....   I don't have time to pursue this quest right now...
The login sequence works when using the Python requests library but not when using the Plex API on my Plex server so it may not be too difficult to fix. Unfortunately the problems to get the streams to work they have with the Kodi plugin and my lack of time leaves me no choice but to pause the plugin.

I uploaded my current messy (not working) code if anyone want to give it a shot!  I
